tests/test-upgrade.py

```diff
@@ -33,6 +33,13 @@ def test_upgrade(self):
         cmd = "sudo snap install microk8s --classic --channel={}".format(upgrade_from).split()
         check_call(cmd)
         wait_for_installation()
+        if in_virtual_env():
+            # In some setups (eg LXC on GCE) the hashsize nf_conntrack file under
+            # sys is marked as rw but any update on it is failing causing kube-proxy
+            # to fail.
+            here = os.path.dirname(os.path.abspath(__file__))
+            apply_patch = os.path.join(here, "patch-kube-proxy.sh")
+            check_call("sudo {}".format(apply_patch).split())
```

Comment:
Do we need this here?  It seems like we would have already done this in `test-distro.sh`, and applying this patch twice would append `--conntrack-max-per-core=0` to kube-proxy args multiple times.

Reply:
You are right that it seems we are reapplying the patch. The real issue it that `test-distro.sh` should also be removing microk8s (not only resetting it) so that the upgrade test will really do the upgrade.